lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -4,50 +4,38 @@ class Pod < KubernetesResource
     TIMEOUT = 10.minutes
     SUSPICIOUS_CONTAINER_STATES = %w(ImagePullBackOff RunContainerError ErrImagePull).freeze
 
-    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+    def initialize(name:, namespace:, context:, file: nil, parent: nil, logger:, deploy_started: nil)
       @name = name
       @namespace = namespace
       @context = context
       @file = file
       @parent = parent
+      @deploy_started = deploy_started
       @logger = logger
-    end
 
-    def sync
-      out, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
-      if @found = st.success?
-        pod_data = JSON.parse(out)
-        interpret_json_data(pod_data)
-      else # reset
-        @status = @phase = nil
-        @ready = false
-        @containers = []
-      end
-      display_logs if unmanaged? && deploy_succeeded?
+      @containers = []
     end
 
-    def interpret_json_data(pod_data)
-      @phase = (pod_data["metadata"]["deletionTimestamp"] ? "Terminating" : pod_data["status"]["phase"])
-      @containers = pod_data["spec"]["containers"].map { |c| c["name"] }
-
-      if @deploy_started && pod_data["status"]["containerStatuses"]
-        pod_data["status"]["containerStatuses"].each do |status|
-          waiting_state = status["state"]["waiting"] if status["state"]
-          reason = waiting_state["reason"] if waiting_state
-          next unless SUSPICIOUS_CONTAINER_STATES.include?(reason)
-          @logger.warn("#{id} has container in state #{reason} (#{waiting_state['message']})")
-        end
+    def sync(pod_data = nil)
+      if pod_data.blank?
+        raw_json, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
+        pod_data = JSON.parse(raw_json) if st.success?
       end
 
-      if @phase == "Failed"
-        @status = "#{@phase} (Reason: #{pod_data['status']['reason']})"
-      elsif @phase == "Terminating"
-        @status = @phase
-      else
-        ready_condition = pod_data["status"].fetch("conditions", []).find { |condition| condition["type"] == "Ready" }
-        @ready = ready_condition.present? && (ready_condition["status"] == "True")
-        @status = "#{@phase} (Ready: #{@ready})"
+      if pod_data.present?
+        @found = true
+        @containers = pod_data["spec"]["containers"].map { |c| c["name"] }
+        interpret_pod_status_data(pod_data["status"], pod_data["metadata"]) # sets @phase, @status and @ready
+        if @deploy_started
+          log_suspicious_states(pod_data["status"].fetch("containerStatuses", []))
+        end
+      else # reset
+        @found = false
+        @containers = []
+        @phase = @status = nil
+        @ready = false
       end
+      display_logs if unmanaged? && deploy_succeeded?
```

Comment:
Example of this output:

![image](https://user-images.githubusercontent.com/4789493/27107642-31457af0-5067-11e7-8bbf-d1c8f3d1a57c.png)